opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp

```diff
@@ -364,21 +368,29 @@ namespace Opm {
       using vec_size_t = std::vector<std::size_t>;
       EclipseGridLGR() = default;
       EclipseGridLGR(const std::string& self_label, const std::string& father_label_, 
-                     int father_lgr_level, size_t nx, size_t ny, size_t nz,
-                     const vec_size_t& father_lgr_index);
+                     size_t nx, size_t ny, size_t nz, 
+                     vec_size_t father_lgr_index, std::array<int,3> low_fahterIJK_, 
+                     std::array<int,3> up_fahterIJK_);
       ~EclipseGridLGR() = default;
       const vec_size_t& getFatherGlobalID() const;
+      void save(Opm::EclIO::EclOutput&, const std::vector<Opm::NNCdata>&, const Opm::UnitSystem&) const;
+      void save_nnc(Opm::EclIO::EclOutput&) const;      
       void set_lgr_global_counter(std::size_t counter){
         lgr_global_counter = counter;
       }
       const vec_size_t& get_father_global() const{
         return father_global;
-      }                 
+      }
+     void set_hostnum(std::vector<int>&);
+     void set_lgr_refinement(std::vector<double>,std::vector<double>);                 
```

Comment:
> This part I need to keep as a non-const reference because I will modify the value of the vector.

Are you talking about `set_hostnum()`?  If so, that's fine.  I was more thinking about `set_lgr_refinement()` which, currently, is specified as taking two `vector<double>`s by value.  Unless those need to be passed by value, I would suggest passing by reference&ndash;and ideally by reference to `const`&ndash;instead.

Reply:
yes, exactly. perfect!